src/software/test_util/equal_within_tolerance.cpp

```diff
@@ -40,6 +40,22 @@ namespace TestUtil
         }
     }
 
+    ::testing::AssertionResult equalWithinTolerance(const Stadium &stadium1, const Stadium &stadium2,
+                                                    double tolerance)
+    {
+        if (equalWithinTolerance(stadium1.segment().getStart(), stadium2.segment().getStart(), tolerance) &&
+            equalWithinTolerance(stadium1.segment().getEnd(), stadium2.segment().getEnd(), tolerance) &&
+            equalWithinTolerance(stadium1.radius(), stadium2.radius(), tolerance))
```

Comment:
nit: We should probably consider the reverse of the segment as well?! What do you think?

Reply:
Good catch!